src/fldeff_teleport.c

```diff
@@ -0,0 +1,55 @@
+#include "global.h"
+#include "field_effect.h"
+#include "task.h"
+
+extern bool8 is_light_level_1_2_3_or_6(u8 mapType);
+extern void* hm_add_c3_launch_phase_2(void);
+extern void hm_teleport_run_dp02scr(void);
+extern void* gUnknown_0203CEEC;
+extern void* gUnknown_03005DB0;
+extern void sub_808469C(void);
+extern u8 GetCursorSelectionMonId(void);
+extern void sub_817C94C(void);
+extern u8 oei_task_add(void);
+extern void FieldEffectActiveListRemove(u8 id);
+extern void sub_80B7FC8(void);
+extern void SetPlayerAvatarTransitionFlags(u8);
+
+
+// 817C8BC
+bool8 SetUpFieldMove_Teleport()
```

Comment:
We normally use the explicit `(void)` if a function takes no arguments. The exception being `GetMonData` and `GetBoxMonData` because they are intentionally used with the wrong prototype, since `()` technically means the function takes an undefined number of arguments`.